File: code/test_suites/cpp/FunctionBuildTest/src/function_pyphonetics_metaphone.cpp
#include <string>
#include <DoubleMetaphone.h>

class PhoneticProcessor {
public:
    static std::string generate_phonetics(const std::string& word) {
        DoubleMetaphone metaphone;
        if (word.empty()) {
            return "The given string is empty.";
        }
        metaphone.setMaxCodeLen(20);
        return metaphone.encode(word);
    }
};

File: code/generate_strategies/IR(CoT)/java/java/function_pyphonetics_metaphone.java
package com.example;

import org.apache.commons.codec.language.DoubleMetaphone;

public class PhoneticProcessor {
    public String generate_phonetics(String word) {
        class PhoneticProcessor {
            public String generate_phonetics(String word) {
                DoubleMetaphone metaphone = new DoubleMetaphone();
                if (word.equals("")) {
                    return "The given string is empty.";
                }
                
                metaphone.setMaxCodeLen(20);
                return metaphone.encode(word);
            }
        }

        return new PhoneticProcessor().generate_phonetics(word);
    }
}